src/lib/autoinstall/package_searcher.rb

```diff
@@ -31,12 +32,49 @@ def initialize(sections)
       @sections = sections
     end
 
-    # Gets packages that needs to be installed
+    # Gets packages that needs to be installed via RPM supplements.
+    # @return [Hash<String, Array<String>>] Required packages of a section. Returning only
+    #   packages that are not already installed.
+    #   Returning NIL package list if no package has been found.
+    def evaluate_via_rpm
+      package_names = {}
+      log.info "Evaluating needed packages for handling AY-sections via RPM Supplements"
+      log.info "Sections: #{sections}"
+      packages = Y2Packager::Resolvable.find(kind: :package)
+
+      sections.each do |section|
+        # Evaluting which package has the supplement autoyast(<section>)
+        package_names[section] = nil
+        packages.each do |p|
+          p.deps.each do |dep|
+            next if !dep["supplements"] ||
+              !dep["supplements"].match?(/^autoyast\(/)
```

Comment:
Suggestion (1. part):

```suggestion
              !dep["supplements"].match(/^autoyast\((.*)\)/)
```